### ISSUE TITLE
`vv vagrant ssh` hangs, but `vagrant ssh` works fine

### ISSUE THREAD

**Issue by ollietreend** [bradp/vv#144]
Hi,

I'm having a problem with the 'vagrant proxy' convenience functions when trying to connect to my vvv VM via SSH.

If I `cd` into the directory for my vvv machine and run `vagrant ssh`, everything works correctly. However if I use vv's vagrant proxy function and run `vv vagrant ssh` from any other directory, the script runs and then hangs. An SSH connection is never opened, and even `Ctrl + C` doesn't stop the script from running. The only option is to close the prompt and terminate the process.

I don't see any error messages whilst this is happening.

I've prepared [a short screen recording to demonstrate the issue](https://www.youtube.com/watch?v=JvCFHyR31oI).

Does anybody else have this issue?

Thanks

**Comment by bradp** [comment]
That makes total sense that the vagrant proxy doesn't work with 'vagrant ssh'. The way the vagrant proxy is implemented is that it returns all the output of a command once it is finished to the parent command. So doing 'vv vagrant up' will echoing all the response when it is finished. Because vagrant ssh doesn't return an exit code, it will not output to the screen, as it doesn't do anything in real time.

**Comment by ollietreend** [comment]
@bradp Understood. I imagined it'd be something like that.

Is there any way we can create a special rule for when `vv vagrant ssh` is run? So rather than running it via the usual proxy implementation, we could so something like: `cd $vvvDir && vagrant ssh`? Thus killing the `vv` command, but allowing the `vagrant ssh` to take over?

Perhaps it would seem like overkill, but it'd be a really nice feature to have available. `vv vagrant ssh` was literally the second command I ran (`vv vagrant up` being the first).

**Comment by bradp** [comment]
That's where the tricky part comes in. Changing directory and running vagrant ssh will not output anything to the user. Nothing bubbles up to the parent shell, everything happens within the context of the script.

The proxying command basically does do a 'cd $vvv_dir && command' and then captures output, and echoes that out. Vagrant ssh gets tricky because not only do we have to show the output of it, we have to capture user input and send that back in.

I'm not sure the best solution for this, but it would be nice to have for sure.

**Comment by ollietreend** [comment]
I think I've found a solution to the issue of running the `vagrant ssh` command, rather than simply proxying its output like with the other commands.

How about using the `exec` command in bash? [According to the Bash Hackers Wiki](http://wiki.bash-hackers.org/commands/builtin/exec), the `exec` command can be used to replace the shell with a given program – executing it, but not as a new process.

Unless I'm mistaken, that should be the solution we're looking for!
